lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -1260,3 +1273,46 @@ def post(self, request, course_id, rolename):
         context = {'course_discussion_settings': CourseDiscussionSettings.get(course_id)}
         serializer = DiscussionRolesListSerializer(data, context=context)
         return Response(serializer.data)
+
+
+class DiscussionModerationSettingsView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+    Retrieve the available reason codes for editing and closing posts to
+    privileged members.
+
+    **Example Requests**
+
+        GET /api/discussion/v1/reason_codes/
+        {
+            "edit_reason_codes": [
+                {"key": "reason-code", "label": "Human-readable label"},
+            ],
+            "close_reason_codes": [
+                {"key": "reason-code", "label": "Human-readable label"},
+            ],
+        }
+    """
+
+    EDIT_REASON_CODES = getattr(settings, "DISCUSSION_MODERATION_EDIT_REASON_CODES", {})
+    CLOSE_REASON_CODES = getattr(settings, "DISCUSSION_MODERATION_CLOSE_REASON_CODES", {})
+
+    def get(self, request, course_id):
+        """
+        Implements the GET method as described in the class documentation.
+        """
+        course_key = CourseKey.from_string(course_id)
+        return Response(
+            {
+                "enabled": ENABLE_DISCUSSION_MODERATION_REASON_CODES.is_enabled(course_key),
+                "edit_reason_codes": [
+                    {"key": reason_code, "label": label}
+                    for (reason_code, label) in self.EDIT_REASON_CODES.items()
+                ],
+                "close_reason_codes": [
+                    {"key": reason_code, "label": label}
+                    for (reason_code, label) in self.CLOSE_REASON_CODES.items()
+                ],
+            },
+            content_type="application/json",
+        )
```

Comment:
@asadazam93 The original author of this PR made this a separate endpoint, but I'm wondering if it's better to just include this in the existing API endpoint at: `/api/discussion/v1/courses/{course_id}/settings`.

That endpoint also includes only moderator-specific settings such as course division settings. Thee are technically not settings and not modifyable. 

They could also be added to `/api/discussion/v1/courses/{course_id}/` which has discussions settings that are visible to users as well.

Reply:
@xitij2000 Agreed

Reply:
@asadazam93 OK. I'm thinking of moving these to the second API since regular users can access these and that might be useful in some situations. I see not reason to restrict the codes to moderators since regular users will be shown these messages in the end.

Reply:
@asadazam93  I've merged this with the existing PR. If the new approach is OK. I will update the frontend as well.

Reply:
@xitij2000 Sure go ahead. I think in case the api is to be used by regular users as well a second api makes sense.